test/selenium/testlib_avocado/seleniumlib.py

```diff
@@ -218,7 +218,7 @@ def click(self, element):
             try:
                 self._selenium_logging("click", element)
                 if javascript_operations:
-                    self.execute_script("arguments[0].click();", element, fatal=False)
+                    self.driver.execute_script("arguments[0].click();", element)
```

Comment:
what is the reason to use it directly without this wrapper? to avoid have another look, when element failure loop is also here in click method?

Reply:
The reason is that `self.execute_script()` has the wrong semantics. This is a retry loop, so with `fatal=True` you get the wrong behaviour: it does a screenshot and converts to a SeniumJSFailure (which our retry loop doesn't catch). But with `fatal=False` it does not except at all, and just logs and silently discards the error -- then the click will not be done, but the code just goes on. We want a `WebDriverException` if the click fails, but no screenshots etc. -- this is exactly what `self.driver.execute_script()` does.